phabricator.js

```diff
@@ -0,0 +1,157 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use strict';
+
+let parentRevision = null;
+let failedToFetch = false;
+
+async function injectButtons() {
+  let blocks = [];
+  let oldCount = 0;
+  do {
+    await wait(1000);
+    oldCount = blocks.length;
+    blocks = document.querySelectorAll('div[data-sigil=differential-changeset]');
+  } while(!blocks.length || oldCount != blocks.length);
+
+  blocks.forEach(block => {
+    let path = block.querySelector('h1.differential-file-icon-header').textContent;
+    let buttonDiv = block.querySelector('div.differential-changeset-buttons');
+    let lines = block.querySelectorAll('table.differential-diff tbody tr th:first-child');
+
+    let button = document.createElement('button');
+    button.type = 'button';
+    button.textContent = 'Code Coverage';
+    button.className = 'button button-grey';
+
+    let enabled = false;
+    let data = null;
+    async function toggle() {
+      enabled = !enabled;
+      if (enabled) {
+        if (data == null) {
+          while (!parentRevision && !failedToFetch) {
+            await wait(1000);
```

Comment:
Instead of this, we should make parentRevision a promise, so we can continue as soon as it's resolved rather than waiting on timeouts.